src/parachains/zombienet.rs

```diff
@@ -515,3 +515,419 @@ impl Source {
+		assert_eq!(binary_system_chain.version, TESTING_POLKADOT_VERSION);
+		assert_eq!(binary_system_chain.sources.len(), 1);
+
+		temp_dir.close()?;
+		Ok(())
+	}
+
+	#[tokio::test]
+	async fn test_parachain() -> Result<()> {
+		//cache
+		let temp_dir = tempfile::tempdir().expect("Could not create temp dir");
+		let cache = PathBuf::from(temp_dir.path());
+
+		let url = Url::parse("https://github.com/r0gue-io/pop-node")?;
+
+		let binary_system_chain = Zombienet::parachain(url, &cache)?;
+
+		assert_eq!(binary_system_chain.name, "pop-node");
+		assert_eq!(binary_system_chain.path, temp_dir.path().join("pop-node"));
+		assert_eq!(binary_system_chain.version, "");
+		assert_eq!(binary_system_chain.sources.len(), 1);
+
+		temp_dir.close()?;
+		Ok(())
+	}
+
+	#[tokio::test]
+	async fn test_missing_binaries() -> Result<()> {
+		//cache
+		let temp_dir = tempfile::tempdir().expect("Could not create temp dir");
+		let cache = PathBuf::from(temp_dir.path());
+
+		let zombienet = Zombienet::new(
+			cache.clone(),
+			CONFIG_FILE_PATH,
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&vec!["https://github.com/r0gue-io/pop-node".to_string()]),
+		)
+		.await?;
+
+		let missing_binaries = zombienet.missing_binaries();
+		assert_eq!(missing_binaries.len(), 3);
+
+		temp_dir.close()?;
+		Ok(())
+	}
+
+	#[tokio::test]
+	async fn test_missing_binaries_no_missing() -> Result<()> {
+		//cache
+		let temp_dir = tempfile::tempdir().expect("Could not create temp dir");
+		let cache = PathBuf::from(temp_dir.path());
+
+		// Create "fake" binary files
+		let relay_chain_file_path = temp_dir.path().join(POLKADOT_BINARY);
+		File::create(relay_chain_file_path)?;
+		let system_chain_file_path = temp_dir.path().join(POLKADOT_PARACHAIN_BINARY);
+		File::create(system_chain_file_path)?;
+		let pop_file_path = temp_dir.path().join("pop-node");
+		File::create(pop_file_path)?;
+
+		let zombienet = Zombienet::new(
+			cache.clone(),
+			CONFIG_FILE_PATH,
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&vec!["https://github.com/r0gue-io/pop-node".to_string()]),
+		)
+		.await?;
+
+		let missing_binaries = zombienet.missing_binaries();
+		assert_eq!(missing_binaries.len(), 0);
+
+		temp_dir.close()?;
+		Ok(())
+	}
+
+	#[tokio::test]
+	async fn test_configure() -> Result<()> {
+		//cache
+		let temp_dir = tempfile::tempdir().expect("Could not create temp dir");
+		let cache = PathBuf::from(temp_dir.path());
+
+		let mut zombienet = Zombienet::new(
+			cache.clone(),
+			CONFIG_FILE_PATH,
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&TESTING_POLKADOT_VERSION.to_string()),
+			Some(&vec!["https://github.com/r0gue-io/pop-node".to_string()]),
+		)
+		.await?;
+
+		let config = zombienet.configure();
+		assert!(config.is_ok());
+
+		temp_dir.close()?;
+		Ok(())
+	}
+
+	#[tokio::test]
```

Comment:
Would be nice to add a comment on what this test is spawning or rename the function to indicate that